features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java

```diff
@@ -435,41 +435,47 @@ private WebApplicationException getException(final Status status, final String m
     }
 
     private DeviceConfigDTO createDeviceConfigDto(DeviceConfigQueryResult queryResult) {
-        Pair<String,String> pair = configToText(queryResult.getEncoding(), queryResult.getConfig());
-        final String encoding = pair.getLeft();
-        final String config = pair.getRight();
-
-        var dto = new DeviceConfigDTO(
-            queryResult.getId(),
-            queryResult.getMonitoredServiceId(),
-            queryResult.getIpAddr(),
-            queryResult.getCreatedTime(),
-            queryResult.getLastUpdated(),
-            queryResult.getLastSucceeded(),
-            queryResult.getLastFailed(),
-            encoding,
-            queryResult.getConfigType(),
-            queryResult.getFilename(),
-            config,
-            queryResult.getFailureReason(),
-            queryResult.getServiceName()
-        );
-
-        // determine backup status, not handling all cases for now
-        boolean backupSuccess = determineBackupSuccess(queryResult.getLastSucceeded(), queryResult.getLastUpdated());
-        dto.setIsSuccessfulBackup(backupSuccess);
-        dto.setBackupStatus(backupSuccess ? BACKUP_STATUS_SUCCESS : BACKUP_STATUS_FAILED);
-
-        dto.setIpInterfaceId(queryResult.getIpInterfaceId());
-        dto.setNodeId(queryResult.getNodeId());
-        dto.setNodeLabel(queryResult.getNodeLabel());
-        dto.setDeviceName(queryResult.getNodeLabel());
-        dto.setLocation(queryResult.getLocation());
-        dto.setOperatingSystem(queryResult.getOperatingSystem());
 
-        populateScheduleInfo(dto);
+        try {
```

Comment:
This seems like a workaround for something that 1) I don't understand and 2) needs more appropriate fixing.

Reply:
Encountered when I removed nodes but device config still exist for those nodes.
This may be covered by https://issues.opennms.org/browse/NMS-14038  but we can keep this for not exposing lower level exceptions in Rest API.